### ISSUE TITLE
FIR CUDA verification fails when  --num-data-per-block > 1024

### ISSUE THREAD

**Issue by apanqasem** [NUCAR-DEV/Hetero-Mark#64]
FIR CUDA verification fails when num-data-per-block > 1024. Basically, it fails for all but one of the input configurations given. Perhaps FP precision issue?

```(capi)% ./fir_cuda  -y 1024 -x 1024 -q -v
CPU: 0, GPU: 0.0594506, both: 0
CPU: 0, GPU: 0.0914504, both: 0
Passed! 1048576 data points filtered
(capi)% ./fir_cuda  -y 1024 -x 2048 -q -v
CPU: 0, GPU: 0.0346238, both: 0
CPU: 0, GPU: 0.0680569, both: 0
At position 1290590, expected 154869552.000000, but was 154869568.000000.
(capi)% ./fir_cuda  -y 1024 -x 8192 -q -v
CPU: 0, GPU: 0.0460944, both: 0
CPU: 0, GPU: 0.0875899, both: 0
At position 1290590, expected 154869552.000000, but was 154869568.000000.
(capi)% ./fir_cuda  -y 8192 -x 8192 -q -v
CPU: 0, GPU: 0.382989, both: 0
CPU: 0, GPU: 0.723651, both: 0
At position 1290590, expected 154869552.000000, but was 154869568.000000.
```
Failure occurs for CUDA only. HC, OpenCL and HIP works fine. We are running codes on Intel Xeon CPU + Nvidia Titan X; Power 8 + Nvidia K40 and Intel Xeon CPU + Nvidia Quadro.

**Comment by syifan** [comment]
@apanqasem Thank you very much for identifying the problem. I can recreate the problem on my platform. It should be caused by the accuracy limitation of 32-bit floating point numbers.  I have fixed in #65. Could you please rerun the program using the branch fix_fir_cuda and verify if the problem has been solved?

**Comment by apanqasem** [comment]
@syifan Thank you. The fix worked!
